specs/Extensions.md

```diff
@@ -0,0 +1,591 @@
+                {
+                    try
+                    {
+                        await extensionsList[i].RemoveAsync();
+                    }
+                    catch (Exception exception)
+                    {
+                        MessageBox.Show("Failed to remove extension: " + exception);
+                    }
+                    found = true;
+                    break;
+                }
+            }
+            if (!found)
+            {
+                MessageBox.Show("Failed to find extension");
+            }
+        }
+        await FillViewAsync();
+    }
+}
+
+```
+# API Notes
+
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+## Win32 C++
+
+```IDL
+interface ICoreWebView2Profile6;
+interface ICoreWebView2EnvironmentOptions6;
+interface ICoreWebView2BrowserExtensionSetEnabledCompletedHandler;
+interface ICoreWebView2BrowserExtensionRemoveCompletedHandler;
+interface ICoreWebView2BrowserExtension;
+interface ICoreWebView2BrowserExtensionList;
+interface ICoreWebView2ProfileAddBrowserExtensionCompletedHandler;
+interface ICoreWebView2ProfileGetBrowserExtensionsCompletedHandler;
+
+/// Additional options used to create WebView2 Environment.
+[uuid(4B1F63E9-F7A5-4EA5-8D84-2B30F2404E82), object, pointer_default(unique)]
+interface ICoreWebView2EnvironmentOptions6 : ICoreWebView2EnvironmentOptions5 {
+  /// When `AreBrowserExtensionsEnabled` is set to `TRUE`, new extensions can be added to user profile and used.
+  /// `AreBrowserExtensionsEnabled` is default to be `FALSE`, in this case, new extensions can't be installed, and
+  /// already installed extension won't be avaliable to use in user profile.
+  /// See `ICoreWebView2BrowserExtension` for Extensions API details.
+  [propget] HRESULT AreBrowserExtensionsEnabled([out, retval] BOOL* value);
+  /// Sets the `AreBrowserExtensionsEnabled` property.
+  [propput] HRESULT AreBrowserExtensionsEnabled([in] BOOL value);
+}
+
+/// This is the ICoreWebView2 profile.
+[uuid(8B16D238-9508-4C36-B4D4-749EB9AC4AD0), object, pointer_default(unique)]
+interface ICoreWebView2Profile6 : IUnknown {
+    /// Adds the [browser extension](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions) using the extension path for unpacked extensions
+    /// from the local device. The extension folder path is the topmost folder of an unpacked browser extension and contains the browser extension manifest file.
+    /// Installed extension will default `IsEnabled` to true.
+    /// When `AreBrowserExtensionsEnabled` is `FALSE`, `AddBrowserExtension` will fail and return HRESULT `ERROR_NOT_SUPPORTED`.
+    /// When an extension is added the extension is persisted in the corresponding profile. The extension will still be installed the next time you use this profile.
+    HRESULT AddBrowserExtension([in] LPCWSTR extensionFolderPath, [in] ICoreWebView2ProfileAddBrowserExtensionCompletedHandler* handler);
```

Comment:
I didn't see the extension get copied to user data folder. 
If after install, the extension script are changed, it will affect the installed extension. The already installed extension won't show up in browser profile anymore.